includes/class-wc-swedbank-plugin.php

```diff
@@ -0,0 +1,554 @@
+				if ( file_exists( $class_file ) ) {
+					  require_once $class_file;
+
+					  return true;
+				}
+
+				return false;
+			}
+		);
+
+		spl_autoload_register(
+			function ( $class_name ) {
+				$auto_load_path = dirname( __FILE__ ) . '/../vendors/swedbank-pay-core-library/src/';
+				$class_file     = $auto_load_path . str_replace( '\\', DIRECTORY_SEPARATOR, $class_name ) . '.php';
+				if ( file_exists( $class_file ) ) {
+					  require_once $class_file;
+
+					  return true;
+				}
+
+				return false;
+			}
+		);
+
+		if ( ! class_exists( '\\Ramsey\\Uuid\\Uuid', false ) ) {
+			require_once $vendors_dir . '/ramsey-uuid/vendor/autoload.php';
+		}
+
+		require_once( dirname( __FILE__ ) . '/class-wc-swedbank-pay-transactions.php' );
+		require_once( dirname( __FILE__ ) . '/class-wc-swedbank-pay-icon.php' );
+	}
+
+	/**
+	 * Install
+	 */
+	public function install() {
+		// Install Schema
+		WC_Swedbank_Pay_Transactions::instance()->install_schema();
+
+		// Set Version
+		if ( ! get_option( 'woocommerce_payex_psp_version' ) ) {
+			add_option( 'woocommerce_payex_psp_version', '1.1.0' );
+		}
+	}
+
+	/**
+	 * Init localisations and files
+	 */
+	public function init() {
+		// Functions
+	}
+
+	/**
+	 * WooCommerce Init
+	 */
+	public function woocommerce_init() {
+		include_once( dirname( __FILE__ ) . '/class-wc-background-swedbank-pay-queue.php' );
+		self::$background_process = new WC_Background_Swedbank_Pay_Queue();
+	}
+
+	/**
+	 * WooCommerce Loaded: load classes
+	 */
+	public function woocommerce_hook_loaded() {
+		// Includes
+		include_once( dirname( __FILE__ ) . '/class-wc-payment-token-swedbank-pay.php' );
+	}
+
+	/**
+	 * Register payment gateway
+	 *
+	 * @param string $class_name
+	 */
+	public static function register_gateway( $class_name ) {
+		global $px_gateways;
+
+		if ( ! $px_gateways ) {
+			$px_gateways = array();
+		}
+
+		if ( ! isset( $px_gateways[ $class_name ] ) ) {
+			// Initialize instance
+			$gateway = new $class_name;
+
+			if ( $gateway ) {
+				$px_gateways[] = $class_name;
+
+				// Register gateway instance
+				add_filter(
+					'woocommerce_payment_gateways',
+					function ( $methods ) use ( $gateway ) {
+						$methods[] = $gateway;
+
+						return $methods;
+					}
+				);
+			}
+		}
+	}
+
+	/**
+	 * Allow processing/completed statuses for capture
+	 *
+	 * @param array $statuses
+	 * @param WC_Order $order
+	 *
+	 * @return array
+	 */
+	public function add_valid_order_statuses( $statuses, $order ) {
+		$payment_method = $order->get_payment_method();
+		if ( in_array( $payment_method, self::PAYMENT_METHODS, true ) ) {
+			$statuses = array_merge(
+				$statuses,
+				array(
+					'processing',
+					'completed',
+				)
+			);
+		}
+
+		return $statuses;
+	}
+
+	/**
+	 * Order Status Change: Capture/Cancel
+	 *
+	 * @param $order_id
+	 * @param $from
+	 * @param $to
+	 * @param $order
+	 */
+	public static function order_status_changed( $order_id, $from, $to, $order ) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid variables with short names like $to. Configured minimum length is 3.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)